### PR TITLE
Apple silicon & Python 3.12

### PR DESCRIPTION
**Pull request by martinellimarco**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        os: [macos-13, ubuntu-latest, windows-latest]
```
Comment (mxmlnkn) [inline .github/workflows/publish.yml:15]: I think you want both: `os: [macos-13, macos-14, ubuntu-latest, windows-latest]`
macos-13 for building for x86-64 and macos-14 for building with ARM. At least that's how I did it in my CI.

Comment (martinellimarco) [reply .github/workflows/publish.yml:15]: I tried macos-14 but for some reason it's not finding zstd.h. It works fine in macos-13 with no other change.
I have CIBW_ARCHS_MACOS: "x86_64 arm64" which seems to be working fine to build both x86-64 and arm64 wheels.
See [this test](https://github.com/martinellimarco/indexed_zstd/actions/runs/8953383213/job/24591832361) under check wheels. Does it seem ok to you?

Comment (mxmlnkn) [reply .github/workflows/publish.yml:15]: Ah nice, looks good and seems more reliable and self-explanatory than running on different macos versions.

Comment (martinellimarco) [reply .github/workflows/publish.yml:15]: Good, I'm going ahead and publishing v1.6.1 :) Let's hope for the best.

**Pull request merged**